src/transformers/modeling_utils.py

```diff
@@ -932,6 +933,18 @@ def prune_heads(self, heads_to_prune: Dict[int, List[int]]):
 
         self.base_model._prune_heads(heads_to_prune)
 
+    def gradient_checkpointing_enable(self, flag: bool = True):
+        """
+        Activates or deactivates gradient checkpointing for the current model.
+
+        Args:
+            flag (:obj:`bool`, `optional`, defaults to :obj:`True`):
+                Will activate gradient checkpointing if :obj:`True`, deactivate it if :obj:`False`.
+        """
+        if not self.supports_gradient_checkpointing and flag:
+            logger.warn(f"{self.__class__.__name__} does not support gradient checkpointing so nothing will happen.")
```

Comment:
any reason not to assert here instead? The user can then change their setup and proceed without problems.

It's a clear error to activate this option if a model doesn't support it, IMHO.

Reply:
It's to be consistent with the previous behavior where we did nothing if the user input `gradient_checkpointing` for a model that did not support it.

I'm not opposed to asserting, but let's see what @LysandreJik and @patrickvonplaten think.

Reply:
Would also be in favor of raising an error here actually. It's a new function so I think we can add this behavior here

Reply:
Will switch then!